### PR TITLE
adding option in the fold builder to invert the fold norm

### PR DESCRIPTION
**Pull request by AngRodrigues**

### CODE REVIEW

File: LoopStructural/modelling/features/fold/_fold.py [hunk 1]
```diff
@@ -99,7 +101,14 @@ def get_deformed_orientation(self, points):
         dgz = np.zeros_like(dgx)
         dgz[:] = np.nan
         dgz[mask, :] = np.cross(dgx[mask, :], fold_axis[mask, :], axisa=1, axisb=1)
-        # dgz = self.foldframe.features[2].evaluate_gradient(points)
+
+        if self.invert_norm == True:
+            dgz = self.foldframe.features[2].evaluate_gradient(points)
```
Comment (lachlangrose) [inline LoopStructural/modelling/features/fold/_fold.py:106]: @AngRodrigues, I think dgz calculation should be the same, it should still be the cross product between the fold axis and the first coordinate of the fold frame. The inverting of the norm should be just replacing line 120
`        dgz[mask][d[mask] < 0] = -dgz[mask][d[mask] < 0]
`
with 
```
if self.invert_norm:
     dgz[mask][d[mask] < 0] = -dgz[mask][d[mask] < 0]
```

**Comment by AngRodrigues** [comment]
Thanks for reviewing this, @lachlangrose. 
Code changed according to comments, in a3641d64d9a3453e941910518c27511fd38dfced 
Let me know if further edits are needed.

File: LoopStructural/modelling/features/fold/_fold.py (hunk 1 above)
Comment (AngRodrigues) [reply LoopStructural/modelling/features/fold/_fold.py]: @lachlangrose 
After closer inspection, I realized that a3641d64d9a3453e941910518c27511fd38dfced wasn't actually saving the inverted values of the norm and returning them.
For some reason, line 120 wasn't updating ```dgz```, but rather the original values were kept. 

Tried to fix this with a3641d64d9a3453e941910518c27511fd38dfced, but not sure if it's a good approach to control the function return with an ```if``` statement?

Comment (lachlangrose) [reply LoopStructural/modelling/features/fold/_fold.py]: @AngRodrigues, no worries. 

I would write it as 

```
if self.invert_norm == True:
            dgz = -dgz[mask][d[mask] <0]

```

and then use a single return, I think it looks nicer but it does the same thing. I wouldn't have the else statement as it is a redundant and if invert norm is not boolean it should not run. invert_norm should be default false and then be set to true, you could check the type and raise a type error if its not boolean or can't be cast to a boolean. 

We should add a test in to check that it works and also check if it catches incorrect types as expected.

**Review by lachlangrose — APPROVED** [review]
All good, @AngRodrigues sorry I missed the updates after my comment will merge now

**Pull request merged**